src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -244,18 +244,20 @@ private async Task<FlagEvaluationDetails<T>> EvaluateFlagAsync<T>(
                 evaluationContextBuilder.Build()
             );
 
-            FlagEvaluationDetails<T> evaluation;
+            FlagEvaluationDetails<T>? evaluation = null;
             try
             {
                 var contextFromHooks = await this.TriggerBeforeHooksAsync(allHooks, hookContext, options, cancellationToken).ConfigureAwait(false);
 
                 // short circuit evaluation entirely if provider is in a bad state
                 if (provider.Status == ProviderStatus.NotReady)
                 {
+                    evaluation = new FlagEvaluationDetails<T>(flagKey, defaultValue, ErrorType.ProviderNotReady, Reason.Error, string.Empty, "Provider has not yet completed initialization.");
                     throw new ProviderNotReadyException("Provider has not yet completed initialization.");
                 }
                 else if (provider.Status == ProviderStatus.Fatal)
                 {
+                    evaluation = new FlagEvaluationDetails<T>(flagKey, defaultValue, ErrorType.ProviderFatal, Reason.Error, string.Empty, string.Empty);

```

Comment:
Won't this be overridden in the catch block?

Reply:
You are right! Removed here: https://github.com/open-feature/dotnet-sdk/pull/335/commits/5cedb628a7b8ac6608801bc13aef18a532dd7c9f